src/dataflow/src/render/reduce.rs

```diff
@@ -28,20 +31,26 @@ use repr::{Datum, DatumList, Row, RowArena, RowPacker};
 use super::context::Context;
 use crate::render::context::Arrangement;
 
-// This enum indicates to the collation operator what results correspond to what types of reductions. Need
-// to keep all of the fused results ahead of the unfused results so that they can be extracted out efficiently
-// The code also currently relies on FusedAccumulable being less than FusedMax.
+// This enum indicates what class of reduction each aggregate function is.
 #[derive(Clone, Debug, Deserialize, Eq, Hash, Ord, PartialEq, PartialOrd, Serialize)]
 enum ReductionType {
-    FusedAccumulable = 1,
-    FusedHierarchical = 2,
-    Unfused = 3,
+    // Accumulable functions can be subtracted from (are invertible), and associative.
+    // We can compute these results by moving some data to the diff field under arbitrary
+    // changes to inputs. Examples include sum or count.
+    Accumulable = 1,
+    // Hierarchical functions are associative, which means we can split up the work of
+    // computing them across subsets. Examples include min or max.
```

Comment:
Minor, but I think an important-ish characteristic is that in addition to being associative, we expect data reduction when the function is called. For example, a hypothetical `sorted_agg` function could be made associative, but there wouldn't be a reason to make it hierarchical. Possibly out of scope for this moment in the code (though perhaps these tags really mean to signify which implementation we'll use).

Reply:
I think I agree with that. I can edit the comment later